protocols/imap/src/main/java/org/apache/james/imap/api/process/ImapSession.java

```diff
@@ -234,9 +233,11 @@ default Username getUserName() {
             .orElse(null);
     }
 
-    default boolean isPlainAuthDisabledEventually() {
-        boolean isPlainAuthDisabled = !isPlainAuthEnabled();
-        boolean isPlainAuthDisabledWhenNotEncrypted = isPlainAuthDisallowed() && !isTLSActive();
-        return isPlainAuthDisabled || isPlainAuthDisabledWhenNotEncrypted;
+    default boolean isPlainAuthDisallowed() {
+        return !isPlainAuthEnabled() || isAuthenticatingNonEncryptedWhenRequiredSSL();
+    }
+
+    default boolean isAuthenticatingNonEncryptedWhenRequiredSSL() {
```

Comment:
I expose this to reuse for OIDC case also. Does this make you more clear, @chibenwa ?